extensions-core/azure-extensions/src/test/java/org/apache/druid/storage/azure/AzureDataSegmentKillerTest.java

```diff
@@ -321,58 +314,69 @@ public void killBatchTest() throws SegmentLoadingException, BlobStorageException
 
     verifyAll();
 
-    Assert.assertEquals(
-        ImmutableSet.of(BLOB_PATH, BLOB_PATH_2),
-        new HashSet<>(deletedFilesCapture.getValue())
-    );
+    assertEquals(ImmutableSet.of(BLOB_PATH, BLOB_PATH_2), new HashSet<>(deletedFilesCapture.getValue()));
   }
 
-  @Test(expected = RuntimeException.class)
+  @Test
   public void test_killBatch_runtimeException()
-      throws SegmentLoadingException, BlobStorageException
   {
-
     EasyMock.expect(azureStorage.batchDeleteFiles(CONTAINER_NAME, ImmutableList.of(BLOB_PATH, BLOB_PATH_2), null))
             .andThrow(new RuntimeException(""));
 
     replayAll();
 
-    AzureDataSegmentKiller killer = new AzureDataSegmentKiller(segmentConfig, inputDataConfig, accountConfig, azureStorage, azureCloudBlobIterableFactory);
-
-    killer.kill(ImmutableList.of(DATA_SEGMENT, DATA_SEGMENT_2));
+    assertThrows(
+        RuntimeException.class,
+        () -> {
+          AzureDataSegmentKiller killer = new AzureDataSegmentKiller(
+              segmentConfig,
+              inputDataConfig,
+              accountConfig,
+              azureStorage,
+              azureCloudBlobIterableFactory
+          );
+          killer.kill(ImmutableList.of(DATA_SEGMENT, DATA_SEGMENT_2));
+        }
+    );
 
     verifyAll();
   }
 
-  @Test(expected = SegmentLoadingException.class)
+  @Test
   public void test_killBatch_SegmentLoadingExceptionOnError()
-          throws SegmentLoadingException, BlobStorageException
   {
-
     EasyMock.expect(azureStorage.batchDeleteFiles(CONTAINER_NAME, ImmutableList.of(BLOB_PATH, BLOB_PATH_2), null))
             .andReturn(false);
 
     replayAll();
 
-    AzureDataSegmentKiller killer = new AzureDataSegmentKiller(segmentConfig, inputDataConfig, accountConfig, azureStorage, azureCloudBlobIterableFactory);
-
-    killer.kill(ImmutableList.of(DATA_SEGMENT, DATA_SEGMENT_2));
+    assertThrows(
+        SegmentLoadingException.class,
+        () -> {
+          AzureDataSegmentKiller killer = new AzureDataSegmentKiller(
+              segmentConfig,
+              inputDataConfig,
+              accountConfig,
+              azureStorage,
+              azureCloudBlobIterableFactory
+          );
+          killer.kill(ImmutableList.of(DATA_SEGMENT, DATA_SEGMENT_2));
```

Comment:
Including both the statements in the lambda makes it difficult to verify that it is `.kill()` that has to throw the exception.